balancer-js/src/modules/pools/provider.ts

```diff
@@ -0,0 +1,59 @@
+import { BalancerSdkConfig, Pool, PoolModel } from '@/types';
+import { PoolRepository, StaticPoolRepository } from '@/modules/data';
+import { Pools as PoolMethods } from './pools.module';
+import { getNetworkConfig } from '../sdk.helpers';
+import pools_14717479 from '@/test/lib/pools_14717479.json';
+
+/**
+ * Building pools from raw data injecting poolType specific methods
+ */
+export class PoolsProvider {
+  constructor(
+    private config: BalancerSdkConfig,
+    private repository: PoolRepository = new StaticPoolRepository( // TODO: replace with proper default repository e.g. subgraph
```

Comment:
Add a subgraph repository (see subgraphPoolDataService.ts) and use this as default.

Reply:
Done ✅